lib/Interpreter/CIFactory.cpp

```diff
@@ -240,102 +241,103 @@ static bool getUniversalCRTSdkDir(std::string &Path,
   Path = sPath;
   return getWindows10SDKVersion(Path, UCRTVersion);
 }
+#endif
 
-  // Get Visual Studio installation directory.
-static bool getVisualStudioDir(std::string &path) {
-  // First check the environment variables that vsvars32.bat sets.
-  const char* vcinstalldir = getenv("VCINSTALLDIR");
-  if (vcinstalldir) {
-    char *p = const_cast<char *>(strstr(vcinstalldir, "\\VC"));
-    if (p)
-      *p = '\0';
-    path = vcinstalldir;
-    return true;
+static void logSearch(const char* Name, const std::string& Value,
+                      const char* Found = nullptr) {
+  if (Found)
+    llvm::errs() << "Found " << Name << " '" << Value << "' that matches "
+                 << Found << " version\n";
+  else
+    llvm::errs() << Name << " '" << Value << "' not found.\n";
+}
+
+static void trimString(const char* Value, const char* SubStr,
```

Comment:
llvm::StringRef::ltrim does individual and consecutive characters.
This is more like (which does look nicer, but has the overhead of two strlens):

```
llvm::StringRef ValueRef(Value);
Path = ValueRef.substr(0, ValueRef.find(SubStr)).str();
```